tests/unittests/tests-core/tests-core-clist.c

```diff
@@ -348,6 +348,45 @@ static void test_clist_is_empty(void)
     }
 }
 
+static void test_clist_special_cardinality(void)
+{
+    unsigned i = 0;
+    TEST_ASSERT(clist_is_empty(&test_clist));
+    TEST_ASSERT(!clist_exactly_one(&test_clist));
+    TEST_ASSERT(!clist_more_than_one(&test_clist));
+
+    clist_rpush(&test_clist, &tests_clist_buf[i++]);
+
+    TEST_ASSERT(!clist_is_empty(&test_clist));
+    TEST_ASSERT(clist_exactly_one(&test_clist));
+    TEST_ASSERT(!clist_more_than_one(&test_clist));
+
+    while (i < TEST_CLIST_LEN) {
+        clist_rpush(&test_clist, &tests_clist_buf[i++]);
+
+        TEST_ASSERT(!clist_is_empty(&test_clist));
+        TEST_ASSERT(!clist_exactly_one(&test_clist));
+        TEST_ASSERT(clist_more_than_one(&test_clist));
+    }
+    do{
+        TEST_ASSERT(!clist_is_empty(&test_clist));
+        TEST_ASSERT(!clist_exactly_one(&test_clist));
+        TEST_ASSERT(clist_more_than_one(&test_clist));
+
+        clist_lpop(&test_clist);
+    }while (--i>1);
```

Comment:
```suggestion
    } while (--i>1);
```

Reply:
and you are trying to sneak in `--i`, right? ;)